utils/profile-services.ts

```diff
@@ -0,0 +1,217 @@
+import { LimitedProfile } from "@pagopa/io-functions-commons/dist/generated/definitions/LimitedProfile";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { ServiceId } from "@pagopa/io-functions-commons/dist/generated/definitions/ServiceId";
+import {
+  RetrievedServicePreference,
+  ServicesPreferencesModel
+} from "@pagopa/io-functions-commons/dist/src/models/service_preference";
+import { asyncIterableToArray } from "@pagopa/io-functions-commons/dist/src/utils/async";
+import {
+  IResponseErrorNotFound,
+  ResponseErrorFromValidationErrors,
+  ResponseErrorNotFound
+} from "italia-ts-commons/lib/responses";
+import * as e from "fp-ts/lib/Either";
+import * as te from "fp-ts/lib/TaskEither";
+import * as t from "fp-ts/lib/Task";
+import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ServicesPreferencesMode";
+import * as a from "fp-ts/lib/Array";
+import { IResponseErrorValidation } from "@pagopa/ts-commons/lib/responses";
+import { BlockedInboxOrChannelEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/BlockedInboxOrChannel";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+
+const NOT_FOUND_TITLE = "ServicesPreferences not found";
+
+interface IServicePreferenceHandler {
+  readonly isMyReposability: (profile: RetrievedProfile) => boolean;
+  readonly handleProfile: (
+    profile: RetrievedProfile,
+    servicesPreferencesModel: ServicesPreferencesModel,
+    serviceId: ServiceId
+  ) => t.Task<LimitedProfile>;
+}
+
+const missingServicePreference = ResponseErrorNotFound(
+  NOT_FOUND_TITLE,
+  "The ServicesPreferences you requested was not found in the system."
+);
+
+/**
+ * Converts the RetrievedProfile model to LimitedProfile type.
+ */
+export const retrievedProfileToLimitedProfile = (
+  retrivedProfile: RetrievedProfile,
+  senderAllowed: boolean
+): LimitedProfile => ({
+  preferred_languages: retrivedProfile.preferredLanguages,
+  sender_allowed: senderAllowed
+});
+
+/**
+ * Whether the sender service is allowed to send
+ * messages to the user identified by this profile
+ */
+export const isSenderAllowed = (
+  blockedInboxOrChannels:
+    | RetrievedProfile["blockedInboxOrChannels"]
+    | undefined,
+  serviceId: ServiceId
+): boolean =>
+  blockedInboxOrChannels === undefined ||
+  blockedInboxOrChannels[serviceId] === undefined ||
+  blockedInboxOrChannels[serviceId].indexOf(BlockedInboxOrChannelEnum.INBOX) <
+    0;
+
+const findServicePreference = (
+  servicesPreferencesModel: ServicesPreferencesModel,
+  profile,
+  serviceId: ServiceId
+): te.TaskEither<IResponseErrorNotFound, RetrievedServicePreference> =>
```

Comment:
I failed to understand the rationale of this method. Are we querying the database for a specific preference of a user? If so, can we use direct access via document id using `find` and `makeServicesPreferencesDocumentId`

Reply:
I think you should report this here https://github.com/pagopa/io-functions-services/pull/136/files#diff-4e0957bac5b9ec17e7bb60a713b90d20a27ce51b0003541ac27fb476b911158b